js/src/lib/components/InferenceWidget/shared/WidgetRealtimeRecorder/Recorder.ts

```diff
@@ -0,0 +1,91 @@
+export default class Recorder {
+	// see developers.google.com/web/updates/2016/01/mediarecorder
+	type: "audio" | "video" = "audio";
+	private apiToken: string | undefined;
+	private audioContext: AudioContext;
+	private isLoggedIn = false;
+	private modelId: string;
+	private onError: (err: string) => void;
+	private renderText: (txt: string) => void;
+	private renderWarning: (warning: string) => void;
+	private socket: WebSocket;
+	private stream: MediaStream;
+
+	constructor(modelId: string, apiToken: string, renderText: (txt: string) => void, renderWarning: (warning: string) => void, onError: (err: string) => void){
+		this.modelId = modelId;
+		this.apiToken = apiToken;
+		// TODO: for testing purposes, supply your hf.co/settings/tokens value in the line below
+		this.apiToken = "";
+		this.renderText = renderText;
+		this.renderWarning = renderWarning;
+		this.onError = onError;
+	}
+
+	async start() {
+		if(!this.apiToken){
+			throw new Error("You need to be loggedn in and have API token enabled. Find more at: hf.co/settings/token");
+		}
+
+		const constraints: MediaStreamConstraints =
+			this.type === "video"
+				? { audio: true, video: true }
+				: { audio: true };
+		this.stream = await navigator.mediaDevices.getUserMedia(constraints);
+
+		this.socket = new WebSocket(`wss://api-inference.huggingface.co/asr/live/cpu/${this.modelId}`);
```

Comment:
using `cpu` all the time for now
read more [here](https://github.com/huggingface/huggingface_hub/pull/594#issuecomment-1018937577)

Reply:
CPU is good enough for now I think!